newarm (v1)

duane701 opened the pull request:
Replace the old (intake) arm with the new intake arm.

Commit:
Fix include paths

WPILIB include path now point to a local path variable `HOME`,
where `HOME` is the local path to the user directory. E.g `C:\Users\Vanden`

Commit:
Change to new intake

The new intake is a single motor with mecanum wheels to draw in the ball.
Update the intake from two motors to one motor.
Rename some functions and parameters to match the new intake.

Commit:
Remove the old Arms subsystem and related commands

The old Arms subsystem is no longer needed with the new design.

Remove the Arms subsystem as well as the related commands that no longer apply.

Commit:
Set up for new arm and controls

duanem committed:
Remove unused commands

More commands that are no longer needed.

duanem committed:
Commands for the new arm

The new arm needs new commands.

Add new commands to control the new arm (intake).

Diego-Andrade commented on src/Commands/SetShifter.cpp:
```diff
@@ -1,41 +1,39 @@
+: mShifterValue(shifterValue)
 {
 	// Use Requires() here to declare subsystem dependencies
 	// eg. Requires(chassis);
```
Requires is required!

Diego-Andrade commented on src/Commands/ToggleArms.cpp:
```diff
@@ -1,40 +1,42 @@
-bool DefaultShooter::IsFinished()
+bool ToggleArms::IsFinished()
 {
 	return false;
```
Going to keep toggling, should be true

Diego-Andrade commented on src/Commands/SetShifter.h:
```diff
@@ -0,0 +1,22 @@
+#ifndef SetShifter_H
+#define SetShifter_H
+
+#include "Commands/Subsystem.h"
```
Should include command.h, not subsystem.h

Diego-Andrade commented on src/Commands/SetArms.h:
```diff
@@ -0,0 +1,22 @@
+#ifndef SetArms_H
+#define SetArms_H
+
+#include "Commands/Subsystem.h"
```
Should include command.h not subsystem.h

Diego-Andrade commented on src/Commands/ToggleArms.h:
```diff
@@ -1,13 +1,13 @@
+#ifndef ToggleArms_H
+#define ToggleArms_H
 
 #include "Commands/Subsystem.h"
```
Should include command.h not substem.h

Diego-Andrade commented on src/Robot.h:
```diff
@@ -37,7 +36,6 @@ class Robot : public IterativeRobot {
   static std::shared_ptr<Shooter> shooter;
```
Fix spacing

Diego-Andrade reviewed (changes requested):
FTP

Diego-Andrade commented on src/Commands/IntakeOn.cpp:
```diff
@@ -1,9 +1,8 @@
 {
-	mHorizontalSpeed = horizontalSpeed;
-	mVerticalSpeed = verticalSpeed;
+	mSpeed = Speed;
```
Use member initialization list

Commit:
Response to original PR

-fixed requires
-fixed member initialization
-fixed tabs
-fixed isFinished returns for toggles and sets
-slight clean

Diego-Andrade commented on src/OI.cpp:
```diff
@@ -21,11 +9,8 @@
 OI::OI() {
     // Process operator interface input here.
-    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTORS
     driver.reset(new Joystick(0));
```
Do we want a const for driver as well? To be consistant

Diego-Andrade commented on src/OI.cpp:
```diff
@@ -68,14 +53,10 @@ OI::OI() {
-// END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=FUNCTIONS
-
 double OI::getDriverLeftAxis() const {
 	return this->driver->GetRawAxis(1);
```
`this->` is not necessary. Also magic number?

Diego-Andrade commented on src/Robot.cpp:
```diff
@@ -32,10 +21,7 @@ void Robot::RobotInit() {
 	oi.reset(new OI());
```
Spacing not fixed

Diego-Andrade reviewed (changes requested):
Spacing is still off, and some suggestions on having magic numbers

duane701 committed:
Tab/Space Substitution

duane701 committed:
Remove `this`

Accidentally left in as an example.

duane701 commented:
Re: Spacing is still off, and some suggestions on having magic numbers

Magic numbers will be fixed in a future PR.

PTAL

Diego-Andrade commented:
LGTM
